Connection error with result code 2 - Connection Refused: identifier rejected.

Hello,

I'm trying the new openhab internal MQTT broker and so far all my devices are working great.

I can't get the MQTT Daemon to connect though (triple-checked correct credentials):

`
[2018-11-05 18:51:24] Connecting to MQTT broker ...
[2018-11-05 18:51:24] Connection error with result code 2 - Connection Refused: identifier rejected.
`

Looks like the broker is not happy with the client ID, see here: https://jaimyn.com.au/mqtt-connection-failed-status-codes-connack-return-codes/

But the config.ini does not allow to change the client ID. how is it chosen?

Cheers

> I do not have a testing system at hand but surely Moquette writes log lines to the file `openhab.log. Those lines for a working and non-working client would be great

I tested it with the original script:
https://github.com/ThomDietrich/miflora-mqtt-daemon/blob/52e4658b40bacbe0809754ed58effc7759c3de48/miflora-mqtt-daemon.py#L155

Afterwards I looked into the openhab.log file. It says that the client ID can not be empty:
```
2019-01-05 22:55:43.995 [ERROR] [.moquette.spi.impl.ProtocolProcessor] - The MQTT client ID cannot be empty. Username=null
```

If I insert the client ID, as described, the connection will work fine.
```
mqtt_client = mqtt.Client(client_id="my_test_id")
```

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,6 +1,6 @@
 #git+https://github.com/open-homeautomation/miflora.git#egg=miflora
 miflora==0.4
-paho-mqtt==1.3.0
+paho-mqtt==1.4.0
 wheel==0.29.0
 sdnotify==0.3.1
 colorama==0.3.9

